### PR TITLE
Have `make update-*` propose to commit any changes

### PR DESCRIPTION
**Pull request by simondeziel**

### CODE REVIEW

**Review by kadinsayani — APPROVED** [review]
This will be super nice to have! Thanks!

File: Makefile [hunk 1]
```diff
@@ -157,6 +157,12 @@ endif
 	@# Generate spec and exclude package from dependency which causes a 'classifier: unknown swagger annotation "extendee"' error.
 	@# For more details see: https://github.com/go-swagger/go-swagger/issues/2917.
 	swagger generate spec -o doc/rest-api.yaml -w ./lxd -m -x github.com/grpc-ecosystem/grpc-gateway/v2/protoc-gen-openapiv2/options
+	if [ -t 0 ] && ! git diff --quiet -- ./doc/rest-api.yaml; then \
+		read -rp "Would you like to commit swagger YAML changes (Y/n)? " answer; \
+		if [ "$${answer:-y}" = "y" ] || [ "$${answer:-y}" = "Y" ]; then \
+			git commit -sm "doc/rest-api: Refresh swagger YAML" -- ./doc/rest-api.yaml;\
```
Comment (kadinsayani) [inline Makefile:163]: Do we want to also include the `-S` flag for `git commit`?

Comment (simondeziel) [reply Makefile:163]: I'm OK with adding "-S" to all of them (including the older one) but if you do provide `-S` on each commit, you'll likely want to add this to your `~/.gitconfig`:

```
# ~/.gitconfig
[commit]
    gpgsign = true
```

**Review by tomponline — APPROVED** [review]

**Comment by tomponline** [comment]
@simondeziel tests not happy though

**Review by tomponline — APPROVED** [review]

**Pull request merged**